### ISSUE TITLE
Release library in Flink independent version way

### ISSUE THREAD

**Issue by novakov-alexey** [flink-extended/flink-scala-api#164]
**Context**: currently this library is released as 3 separate published JARs for the last 3 Apache Flink versions. For example:

- 1.18.1_1.2.0
- 1.19.1_1.2.0
- 1.20.0_1.2.0

Such approach make the release process quite slow, because it requires 3 bundles to be published to the Sonatype repository.
Also, it forces a user to think which version to select based on his Flink version used in a project. Not a big deal, but why bother with this?

**Proposal**: 
1. Build and publish a single JAR only without Flink version prepended.

2. In order to keep Flink compatibility guarantees, still run sbt tests in GH Action for the last 3 Flink version as it is now.
In the result, this library gets verified across different Flink versions, but will have single version schema, which simplifies dependency management process for the users.

**Caveats**

1. What if build fails in one or two Flink version while it is still successful for one of the versions. This may happen in future with Flink 2.0 and Flink 1.x

**Comment by novakov-alexey** [comment]
@arnaud-daroussin FYI, there is a proposal to release single library version, but still compiled against the last 3 Flink version in CI.

**Comment by arnaud-daroussin** [comment]
Hi @novakov-alexey,

The risk I see in removing Flink versions from the bundle versioning is the inability to manage the breaking changes that will come with Flink 2.0.

By keeping separate builds and bundles, it will be possible to have different sources root for different Flink versions, as is already the case for Scala 2 and 3.

In case of a breaking change with a single bundle, it will be necessary either to:
- no longer create new versions for Flink 1.x
- maintain Flink 1.x on a separate branch, which will require more manual work in the end

In conclusion, I am not sure that this is the right time to make this simplification, and I find that the build time of less than 5 minutes is still acceptable.

**Comment by novakov-alexey** [comment]
Hi @arnaud-daroussin ,

Thanks for your input! 
Separate sources for Scala 2 and 3 will remain, they are not really related to the different Flink versions. They are more to compile Scala 2 and Scala 3 syntax.

Before Flink 2.0 support, we could live with single version of this library. Then we will probably have two versions: API-1.0-xyz and API-2.0-xyz

However, I am not sure how to support Flink 2.0. Will it require to have separate module called api-2.0 which would not depend any more on the removed, previously deprecated, Flink Java methods or not. By the way, as Flink 2.0 preview is already published, we can explore what takes to support it in future.

**Comment by novakov-alexey** [comment]
closed #239 

- we have now two flink-scala-api modules: Flink 1.x and Flink 2.y.
- there is no sub version for different 1.x or 2.y versions.
- however we will still compile and test several Flink versions in the GitHub Pipeline
